src/ImageSharp.Web/Processors/WebProcessingExtensions.cs

```diff
@@ -33,15 +35,32 @@ internal static class WebProcessingExtensions
             CommandParser commandParser,
             CultureInfo culture)
         {
-            if (commands.Count != 0)
+            foreach (IImageWebProcessor processor in processors.GetBySupportedCommands(commands))
             {
-                foreach (IImageWebProcessor processor in processors)
-                {
-                    source = processor.Process(source, logger, commands, commandParser, culture);
-                }
+                source = processor.Process(source, logger, commands, commandParser, culture);
             }
 
             return source;
         }
+
+        /// <summary>
+        /// Sorts the processors according to the first supported command and removes processors without supported commands.
+        /// </summary>
+        /// <param name="processors">The collection of available processors.</param>
+        /// <param name="commands">The parsed collection of processing commands.</param>
+        /// <returns>
+        /// The sorted proccessors.
+        /// </returns>
+        public static IEnumerable<IImageWebProcessor> GetBySupportedCommands(this IEnumerable<IImageWebProcessor> processors, IDictionary<string, string> commands)
+        {
+            List<string> commandKeys = new List<string>(commands.Keys);
+            IEqualityComparer<string> comparer = (commands as Dictionary<string, string>)?.Comparer ?? StringComparer.OrdinalIgnoreCase;
+
+            return processors
+                .GroupBy(p => commandKeys.Intersect(p.Commands, comparer).Min(c => (int?)commandKeys.FindIndex(k => comparer.Equals(c, k))) ?? -1) // Get index of first supported command
```

Comment:
Similar to `ImageSharpMiddleware`, we can also remove the above `comparer` and just use:
```suggestion
                .GroupBy(p => commandKeys.Intersect(p.Commands, StringComparer.OrdinalIgnoreCase).Min(c => (int?)commandKeys.FindIndex(k => k.Equals(c, StringComparison.OrdinalIgnoreCase))) ?? -1) // Get index of first supported command
```

If we do this, we might want to change the `commands` parameter type into a list, as we don't need the dictionary/values in this method anyway...